src/main/java/io/logz/logback/LogzioSender.java

```diff
@@ -307,6 +323,23 @@ private String formatMessage(ILoggingEvent loggingEvent) {
         logMessage.addProperty("logger", loggingEvent.getLoggerName());
         logMessage.addProperty("thread", loggingEvent.getThreadName());
 
+        if (hostname != null) {
+            logMessage.addProperty("hostname", hostname);
+        }
+        if (additionalFields != null) {
+            additionalFields.forEach((k,v) -> {
+                if (v.startsWith("$")) {
+                    String environmentValue = System.getenv(v.replace("$",""));
+                    if (environmentValue != null) {
+                        logMessage.addProperty(k, environmentValue);
+                    }
+                }
+                else {
+                    logMessage.addProperty(k, v);
```

Comment:
Let's make sure you're not overriding existing fields (by mistake like 'message'). Check if the field doesn't exist and WARN on it.